test/shopify_app/session/session_repository_test.rb

```diff
@@ -0,0 +1,162 @@
+require 'test_helper'
+
+module ShopifyApp
+  class SessionRepositoryTest < ActiveSupport::TestCase
+
+    teardown do
+      SessionRepository.shop_storage = nil
+      SessionRepository.user_storage = nil
+    end
+
+    test ".user_storage= raises ArgumentError if the object is missing .store" do
+      storage = Class.new do
+        def retrieve; end
+        def retrieve_by_user_id; end
+      end
+
+      assert_raises(ArgumentError) { SessionRepository.user_storage = storage.new }
+    end
+
+    test ".user_storage= raises ArgumentError if the object is missing .retrieve" do
+      storage = Class.new do
+        def store; end
+        def retrieve_by_user_id; end
+      end
+
+      assert_raises(ArgumentError) { SessionRepository.user_storage = storage.new }
+    end
+
+    test ".user_storage= raises ArgumentError if the object is missing .retrieve_by_user_id" do
+      storage = Class.new do
+        def store; end
+        def retrieve; end
+      end
+
+      assert_raises(ArgumentError) { SessionRepository.user_storage = storage.new }
+    end
+
+    test ".user_storage= does not raise ArgumentError if input is nil" do
+      assert_nothing_raised { SessionRepository.user_storage = nil }
+    end
+
+    test ".user_storage= clears storage setting if input is nil" do
+      SessionRepository.user_storage = InMemoryUserSessionStore
+      assert_not_nil SessionRepository.user_storage
```

Comment:
It'll never be `nil` so we should probably `assert_kind_of InMemoryUserSessionStore.class, SessionRepository.user_storage`